### PR TITLE
docs(docs-site): add motion components page 

### LINKED ISSUE #32411: [Feature]: Dialog stories do not include an example for customizing (or disabling) motion
### Library

React Components / v9 (@fluentui/react-components)

### Describe the feature that you would like added

The dialog documentation / stories does not include information on how to customize or disable animations. The only information is a brief mention of the types for `surfaceMotion`. In fact `backdropMotion` is not even mentioned in the docs. 

### Have you discussed this feature with our team

_No response_

### Additional context

This issue comes from a partner request about "how to disable motion on dialog". 

### Validations

- [X] Check that there isn't already an issue that requests the same feature to avoid creating a duplicate.

### Priority

Normal

### PR DESCRIPTION
**Pull request by mainframev**
Adding additional pages to `react-motion` with `motion slot` examples. Added `hideArgsTable` docs parameter to hide `ArgsTable` block. 

## New Behavior

Added motion components page and links to related components. 

## Related Issue(s)

- Fixes #32411

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Review by layershifter — CHANGES_REQUESTED** [review]
See https://github.com/microsoft/fluentui/issues/32411#issuecomment-2327197175.

IMO adding examples for the same concept across components - does not scale 🥲

 We should have a shared documentation page that explains how customize/disable motion (see docs for `positioning`) and link to it components that are using it.

**Comment by mainframev** [comment]
> See [#32411 (comment)](https://github.com/microsoft/fluentui/issues/32411#issuecomment-2327197175).
> 
> IMO adding examples for the same concept across components - does not scale 🥲
> 
> We should have a shared documentation page that explains how customize/disable motion (see docs for `positioning`) and link to it components that are using it.

@layershifter makes sense. Should I create a separate request for it or can do within current? wdyt?

**Comment by layershifter** [comment]
> @layershifter makes sense. Should I create a separate request for it or can do within current? wdyt?

Up to you 🐱

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">41</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">34</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/Avatar-mount_base.html">661</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/Avatar-mount.html">635</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/Button-mount_base.html">305</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/Button-mount.html">301</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/Field-mount_base.html">1153</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/Field-mount.html">1177</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProvider-mount_base.html">725</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProvider-mount.html">717</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProviderWithTheme-mount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProviderWithTheme-mount.html">86</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">41</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">34</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">81</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">78</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/MakeStyles-mount_base.html">865</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/MakeStyles-mount.html">872</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/Persona-mount_base.html">1828</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/Persona-mount.html">1750</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/SpinButton-mount_base.html">1413</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/SpinButton-mount.html">1411</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/SwatchPicker-mount_base.html">1654</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32437/perf-test-react-components/SwatchPicker-mount.html">1666</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: apps/public-docsite-v9/src/Concepts/Motion/CustomMotion.stories.tsx
Comment (fabricteam) [inline apps/public-docsite-v9/src/Concepts/Motion/CustomMotion.stories.tsx:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32437&commitId=dc7e87e80fbdd1d9f95e6faf3a3cf84c1d8d9a70&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.chromium.png|3| Changed|
|Avatar Converged.Badge Mask RTL.chromium.png|19| Changed|

</details>


  <details><summary><b>Drawer</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.Full Overlay Dark Mode.chromium.png|975| Changed|
|Drawer.Full Overlay High Contrast.chromium.png|2244| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by layershifter — APPROVED** [review]

File: apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx
Comment (Hotell) [inline apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx:144]: - are these official SB apis ? 
- seems this PR does way beyond adding documentation. please update PR and title accordingly

File: packages/react-components/react-motion/stories/src/CreatePresenceComponent/utils.stories.tsx
Comment (Hotell) [inline packages/react-components/react-motion/stories/src/CreatePresenceComponent/utils.stories.tsx:8]: I'm missing the point of this artificial component that does noop, can you provide more context? ( ideally remove  it )

File: packages/react-components/react-drawer/library/src/components/OverlayDrawer/OverlayDrawer.types.ts
Comment (Hotell) [inline packages/react-components/react-drawer/library/src/components/OverlayDrawer/OverlayDrawer.types.ts:18]: will this link actually work ? lets make those links work from both source code and docs

File: packages/react-components/react-motion/stories/src/Motion/index.stories.tsx
Comment (Hotell) [inline packages/react-components/react-motion/stories/src/Motion/index.stories.tsx:8]: all these stories seems to actually go against the patterns we agreed - this and others related in this PR should be a `.mdx` rather than ts which doesn't provide any real `components` https://github.com/microsoft/fluentui/pull/32465#issuecomment-2331293982

File: packages/react-components/react-drawer/library/src/components/OverlayDrawer/OverlayDrawer.types.ts
Comment (Hotell) [reply packages/react-components/react-drawer/library/src/components/OverlayDrawer/OverlayDrawer.types.ts:18]: but it will not work from source right, that needs to be fixed.

also wondering why it wont work on staging ? you dont have there origin URL (yet) the rest should be same no matter where it lives no?

File: packages/react-components/react-motion/stories/src/Motion/index.stories.tsx
Comment (Hotell) [reply packages/react-components/react-motion/stories/src/Motion/index.stories.tsx:8]: have you seen this comment @mainframev  ? I guess not as you re-requested my review and didnt cover what I was asking ?

**Review by Hotell — COMMENTED** [review]
comments from previous review haven't been addressed

File: packages/react-components/react-drawer/library/src/components/OverlayDrawer/OverlayDrawer.types.ts
Comment (mainframev) [reply packages/react-components/react-drawer/library/src/components/OverlayDrawer/OverlayDrawer.types.ts:18]: > also wondering why it wont work on staging ? you dont have there origin URL (yet) the rest should be same no matter where it lives no?

staging url is appended with PR number `/pull/32437` after the baseUrl, which gets overwritten by the relative path to motion doc page. Let's use the absolute full path, that will work as expected everywhere after we merge this

Comment (Hotell) [reply packages/react-components/react-drawer/library/src/components/OverlayDrawer/OverlayDrawer.types.ts:18]: sounds good, lets go with full path , ty

File: packages/react-components/react-motion/stories/src/Motion/CustomMotion.stories.tsx
Comment (Hotell) [inline packages/react-components/react-motion/stories/src/Motion/CustomMotion.stories.tsx:134]: lets inline this, the content is simple enough. with any `import` we create invalid stackblitz demo -> we process and export only source file of story definition

File: packages/react-components/react-motion/stories/src/Motion/DisableMotion.stories.tsx
Comment (Hotell) [inline packages/react-components/react-motion/stories/src/Motion/DisableMotion.stories.tsx:67]: same as previous comment - lets inline

File: apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx
Comment (mainframev) [reply apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx:144]: parameters is official sb api :) they are normally used to define the behavior of some Storybook features or Storybook addons, but also can be used to pass any arbitrary data

File: apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx
Comment (Hotell) [inline apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx:175]: discussed offline: it seems that this is needed only for special purposes of motion stories, which are not adhering to our SB patterns for authoring stories. We should follow our practices and if there is something that needs to change it should be discussed with others and documented(ideally automated). I dunno who introduced these new patterns for motion stories so cc-ing @layershifter ty

**Review by Hotell — APPROVED** [review]
approving to unblock, although please dont forget to:

- please address actionable comments
- regarding the conversion to pure MDX, this will be tackled as a follow up

**Pull request merged**
